packages/flutter_sprites/README.md

```diff
@@ -1,8 +1,209 @@
+The most common node type is the Sprite node. A sprite simply draws an image to the screen. Sprites can be drawn from Image objects or Texture objects. A texture is a part of an Image. Using a SpriteSheet you can pack several texture elements within a single image. This saves space in the device's gpu memory and also make drawing faster. Currently Flutter Sprites supports sprite sheets in json format and produced with a tool such as TexturePacker. You can create a SpriteSheet with a definition in json and an image:
+
+    SpriteSheet sprites = new SpriteSheet(myImage, jsonCode);
+    Texture texture = sprites['texture.png'];
+
+## The frame cycle
+Each time a new frame is rendered to screen Flutter Sprites will perform a number of actions. Sometimes when creating more advanced interactive animations or games, the order in which these actions are performed may matter.
+
+This is the order things will happen:
+
+- Handle input events
+- Run animation actions
+- Call update functions on nodes
+- Apply constraints 
+- Render the frame to screen
+
+Read more about each of the different phases below.
+
+## Animating using actions
+Flutter Sprites provides easy to use functions for animating nodes through actions. You can combine simple action blocks to create more complex animations.
+
+### Tweens
+Tweens are the simplest building block for creating an animation. It will interpolate a value or property over a specified time period. You provide the ActionTween class with a setter function, its start and end value, and the duration for the tween.
+
+After creating a tween, execute it by running it through a node's action manager.
+
+	Node myNode = new Node();
+
+    ActionTween myTween = new ActionTween(
+      (Point a) => myNode.position = a,
+      Point.origin,
+      const Point(100.0, 0.0),
+      1.0
+    );
+    
+    myNode.actions.run(myTween);
+
+You can animate values of different types, such as floats, points, rectangles, and even colors. You can also optionally provide the ActionTween class with an easing function.
+
+### Sequences
+When you need to play two or more actions in a sequence, use the ActionSequence class:
+
+    ActionSequence sequence = new ActionSequence([
+      firstAction,
+      middleAction,
+      lastAction
+    ]);
+    
+### Groups
+Use ActionGroup to play actions in parallel:
+
+    ActionGroup group = new ActionGroup([
+      action0,
+      action1
+    ]);
+    
+### Repeat
+You can loop any action, either a fixed number of times, or until the end of times:
+
+    ActionRepeat repeat = new ActionRepeat(loopedAction, 5);
+    
+    ActionRepeatForever longLoop = new ActionRepeatForever(loopedAction);
+    
+### Composition
+It's possible to create more complex actions by composing them in any way:
+
+    ActionSequence complexAction = new ActionSequence([
+      new ActionRepeat(myLoop, 2),
+      new ActionGroup([
+      	action0,
+      	action1
+      ])
+    ]);
+
+## Constraints
```

Comment:
Please consider renaming the 'Constraints' section to a task-oriented name, e.g. "Defining constraints". Similarly, please consider renaming "Custom drawing" to "Create custom drawings", and "Particle systems" to "Add effects using particle systems".